src/main/java/org/fiware/iam/dsp/RainbowInitializer.java

```diff
@@ -0,0 +1,27 @@
+package org.fiware.iam.dsp;
+
+import io.micronaut.context.event.StartupEvent;
+import io.micronaut.runtime.event.annotation.EventListener;
+import jakarta.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import org.fiware.iam.configuration.GeneralProperties;
+import org.fiware.iam.exception.RainbowException;
+
+@Singleton
+@RequiredArgsConstructor
+public class RainbowInitializer {
+
+	private static final String PROVIDER_ROLE = "Provider";
+
+	private final RainbowAdapter rainbowAdapter;
+	private final GeneralProperties generalProperties;
+
+	@EventListener
+	public void initializeProvider(StartupEvent startupEvent) {
+		rainbowAdapter
+				.isParticipant(generalProperties.getDid())
+				.filter(r -> !r)
+				.flatMap(r -> rainbowAdapter.createParticipant(generalProperties.getDid(), PROVIDER_ROLE))
```

Comment:
Does the rainbow endpoint allow updating the participant too?
Scenario 1: CM restarts and tries to create the participant but fails because it already exists (probably no problem)
Scenario 2: We run a setup where both sides have the whole DSC stack running and might overwrite the config as the same did is used for consumer and provider side inside the same DSC

Reply:
For now, we only use creation. This should probably taken into account in future iterations.